changed gd Formula to using my bugfixed version of libgd

konqui opened the pull request:
See why in the release notes on https://github.com/konqui/libgd/releases/tag/gd-2.1.1-rc1

The other pull request i have closed.
For me it seems like this is the better way for solving the problem.

No extra patch download needed.
And the Formula can easy be change back to the original source at any time when there is a new release of libgd out.

DomT4 commented:
We don't accept changing formulae to alternative forks unless blessed by upstream. You'll need to use a patch here, if there's one available, rather than cutting your own release and moving Homebrew users across to your personal fork.

More generally we don't often permit [authors to submit their own projects](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#niche-or-self-submitted-stuff).

konqui commented:
Ok i can understand that this isn't the right way to feed in the needed changes.

I just saw that the Formula didn't build so i took a look and saw the error and fixed it for me. And dicided to try to give it back to homebrew cause i like to use homebrew and i thought it would be a good idea to give something back.
ich I first opend an issue and added a git-diff patch to it. Then @mikemcquaid said that it would be better to make a pull request to handle in my patch.

I tried but then you decided that it would be the better Way to give it to the Maintainer of libgd it selfs. (i thought this would be a good idea too so i tried it)

I have then given my Patch to the Original Maintainer of libgd but he refused to use/apply it cause he decided to move away from libvpx at all. So no Luck there.

On the other Hand he also hasn't any newer Release out by now. So homebrew is stuck at gd-2.1.1 which has the error in it.

So you see my dilemma now?

I can't change the Formula to use a newer Version of libgd which doesn't fail cause there simply isn't one yet.

My Patch also got declined. So my last atempt was to try to use my own patched version of libgd.

Now you tell me that this also isn't the Propper way.

 I know but I also tried every way i could think of in giving others access to my patch.

How or where am i supposed to handle in the needed changes to libgd if i wanna make them avaliable to others not just me?

mikemcquaid commented:
> I have then given my Patch to the Original Maintainer of libgd but he refused to use/apply it cause he decided to move away from libvpx at all. So no Luck there.
> 
> On the other Hand he also hasn't any newer Release out by now. So homebrew is stuck at gd-2.1.1 which has the error in it.

Thanks for the explanation. In this case that means we'll be happy to accept your patch to make it work with the newer `libgd`. If you can submit the PR to the patches repo and then change this pull request to apply your patch then we will accept that approach.
